src/freertos_drivers/common/DccDecoder.hxx

```diff
@@ -24,34 +24,30 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * \file TivaNRZ.hxx
+ * \file DccDecoder.hxx
  *
- * Device driver for TivaWare to decode DCC track signal.
+ * Generic device driver for decoding DCC track signal using a timer resource
+ * in capture mode.
  *
  * @author Balazs Racz
- * @date 29 Nov 2014
+ * @date 8 April 2020
  */
 
-#include "TivaDCC.hxx"  // for FixedQueue
-#include "TivaGPIO.hxx" // for pin definitions
 #include "RailcomDriver.hxx" // for debug pins
 #include "dcc/Receiver.hxx"
 
-typedef DummyPin PIN_RailcomCutout;
-
 /**
-  Device driver for decoding a DCC signal on a TI Tiva class microcontroller.
- 
-  This driver exports a filesystem device node, but that file node is not
-  usable for reading or writing anything at the moment. The only feature
-  supported by this device driver right now is that it is able to tell a
-  RailcomDriver when the railcom cutout is supposed to start, when we're in the
-  middle and when it is over. This is necessary for the correct functionality
-  of the railcom driver.
-
-  @todo: implement actual packet decoding and sending back to the application
-  layer.
+  Device driver for decoding a DCC signal using a Timer resource.
+
+  This driver exports a filesystem device node, which will be readable for
+  decoded DCC packets (read into struct DCCPacket). All writes fail.
+
+  The additional feature supported by this device driver is that it is able to
+  tell a RailcomDriver when the railcom cutout is supposed to start, when we're
+  in the middle and when it is over. This is necessary for the correct
+  functionality of the railcom driver.
 
+  @todo describe usage.
```

Comment:
Should this still be a todo item? It looks like you are describing usage.

Reply:
Actually all this description was outdated stuff that should have been on the Tiva side. Wrote new description.